llvm/lib/Target/DirectX/DXContainerGlobals.cpp

```diff
@@ -104,6 +112,38 @@ GlobalVariable *DXContainerGlobals::computeShaderHash(Module &M) {
   return GV;
 }
 
+GlobalVariable *DXContainerGlobals::buildContainerGlobal(
+    Module &M, Constant *Content, StringRef Name, StringRef SectionName) {
+  auto *GV = new llvm::GlobalVariable(
+      M, Content->getType(), true, GlobalValue::PrivateLinkage, Content, Name);
+  GV->setSection(SectionName);
+  GV->setAlignment(Align(4));
+  return GV;
```

Comment:
You added this function but it is only called in one place. If you intend this as a refactoring, we should extend it across the file, otherwise it seems odd to add a function that is only called from one location.

Reply:
Updated.